sdk/eventhub/Azure.Messaging.EventHubs/MigrationGuide_WindowsAzureServiceBus.md

```diff
@@ -0,0 +1,692 @@
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producer = new EventHubProducerClient(connectionString, eventHubName);
+
+try
+{
+    using EventDataBatch eventBatch = await producer.CreateBatchAsync();
+
+    for (var index = 0; index < 5; ++index)
+    {
+        var eventData = new EventData($"Event #{ index }");
+
+        if (!eventBatch.TryAdd(eventData))
+        {
+            throw new Exception($"The event at { index } could not be added.");
+        }
+    }
+
+    await producer.SendAsync(eventBatch);
+}
+finally
+{
+    await producer.CloseAsync();
+}
+```
+
+#### Publishing events with a partition key
+
+In the `WindowsAzure.ServiceBus` library, to publish events and allow the Event Hubs service to automatically assign the partition, the `EventHubClient` was used as the publisher for a batch created with a partition key specified as an option.
+
+```C#
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+EventHubClient client = EventHubClient.CreateFromConnectionString(connectionString, eventHubName);
+
+try
+{
+    using var eventBatch = client.CreateBatch();
+
+    for (var index = 0; index < 5; ++index)
+    {
+        var eventData = new EventData(Encoding.UTF8.GetBytes($"Event #{ index }"));
+        eventData.PartitionKey = "Any Value Will Do...";
+
+        if (!eventBatch.TryAdd(eventData))
+        {
+            throw new Exception($"The event at { index } could not be added");
+        }
+    }
+
+    await client.SendBatchAsync(eventBatch.ToEnumerable());
+}
+finally
+{
+    client.Close();
+}
+```
+
+In the `Azure.Messaging.EventHubs` library, the `EventHubProducerClient` is used as the publisher for a batch created with a partition key specified as an option.
+
+```C# Snippet:EventHubs_Sample04_PartitionKey
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producer = new EventHubProducerClient(connectionString, eventHubName);
+
+try
+{
+    var batchOptions = new CreateBatchOptions
+    {
+        PartitionKey = "Any Value Will Do..."
+    };
+
+    using EventDataBatch eventBatch = await producer.CreateBatchAsync(batchOptions);
+
+    for (var index = 0; index < 5; ++index)
+    {
+        var eventData = new EventData($"Event #{ index }");
+
+        if (!eventBatch.TryAdd(eventData))
+        {
+            throw new Exception($"The event at { index } could not be added.");
+        }
+    }
+
+    await producer.SendAsync(eventBatch);
+}
+finally
+{
+    await producer.CloseAsync();
+}
+```
+
+#### Publishing events to a specific partition
+
+In the `WindowsAzure.ServiceBuss` library, to publish events and allow the Event Hubs service to automatically assign the partition, a `EventHubSender` is created using the `EventHubClient` and then used as the publisher for a batch created with its default configuration.
```

Comment:
...to publish events to a specific, user-specified partition, an 'EventHubSender'...
This scenario is about NOT allowing the service to assign!

Reply:
Thanks @JamesBirdsall!

Fixed both here and in the other migration guide: https://github.com/Azure/azure-sdk-for-net/pull/26219/commits/19f9ffee89054a2ebbd03da9da68ff72de36f0f1